components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.ts

```diff
@@ -0,0 +1,110 @@
+import { set, pipe } from 'lodash/fp';
+import { LoadingStatus } from 'app/types/types';
+import {
+  ProjectsFilterActions,
+  ProjectsFilterActionTypes
+} from './projects-filter.actions';
+
+export interface ProjectsFilterOption {
+  label: string;
+  value: string;
+  checked: boolean;
+}
+
+export interface ProjectsFilterState {
+  options: ProjectsFilterOption[];
+  optionsLoadingStatus: LoadingStatus;
+  selectionLabel: string;
+  selectionCount: number;
+  selectionCountVisible: boolean;
+  selectionCountActive: boolean;
+}
+
+export const projectsFilterInitialState: ProjectsFilterState = {
+  options: [],
+  optionsLoadingStatus: LoadingStatus.notLoaded,
+  selectionLabel: 'All resources',
+  selectionCount: 0,
+  selectionCountVisible: false,
+  selectionCountActive: false
+};
+
+export function projectsFilterReducer(
+  state: ProjectsFilterState = projectsFilterInitialState,
+  action: ProjectsFilterActions): ProjectsFilterState {
+
+  switch (action.type) {
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS: {
+      return set('optionsLoadingStatus', LoadingStatus.loading, state);
+    }
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS_SUCCESS: {
+      return pipe(
+        set('options', action.payload),
+        set('optionsLoadingStatus', LoadingStatus.loadingSuccess),
+        set('selectionLabel', selectionLabel(action.payload)),
+        set('selectionCount', selectionCount(action.payload)),
+        set('selectionCountVisible', selectionCountVisible(action.payload)),
+        set('selectionCountActive', selectionCountActive(action.payload))
+      )(state) as ProjectsFilterState;
+    }
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS_FAILURE: {
+      return set('optionsLoadingStatus', LoadingStatus.loadingFailure, state);
+    }
+
+    case ProjectsFilterActionTypes.SAVE_OPTIONS: {
+      return pipe(
+        set('options', action.payload),
+        set('selectionLabel', selectionLabel(action.payload)),
+        set('selectionCount', selectionCount(action.payload)),
+        set('selectionCountVisible', selectionCountVisible(action.payload)),
+        set('selectionCountActive', selectionCountActive(action.payload))
+      )(state) as ProjectsFilterState;
+    }
+  }
+
+  return state;
+}
+
+function selectionLabel(options: ProjectsFilterOption[]): string {
```

Comment:
In this function and elsewhere it looks like you've implemented the line items under "Access to more than 1 project and not the unassigned resources" from A2-248 but not the "Access to at least 1 project and the unassigned resources".

(And uncommenting "Unassigned Resources" in the sample data seems to confirm that.)

Reply:
@msorens that's still a todo. I wasn't sure how y'all were going to want to include 'unassigned resources' with all the other projects options available to a user. Will it have a special ID or some other flag to differentiate it from projects?

Reply:
Eh, needed `if` statements will be the same regardless of what y'all decide so I'll go ahead and tweak this code.

Reply:
https://github.com/chef/automate/pull/172/commits/8c9c813f05108db925930ff535cee3a45a48dba6